cpmpy/transformations/linearize.py

```diff
@@ -572,3 +604,26 @@ def only_positive_coefficients(lst_of_expr):
 
     return newlist
 
+def linearize_objective(expr, supported=frozenset(["sum","wsum"])):
+    """
+    - Decision variable: Var
+    - Linear: sum([Var])                                   (CPMpy class 'Operator', name 'sum')
+              wsum([Const],[Var])                          (CPMpy class 'Operator', name 'wsum')
+              no negated BoolVars
+
+    Only difference with flatten_objective is that no negated boolvars are allowed.
+    """
+    flatexpr, flatcons = flatten_objective(expr, supported=supported)
+    pos_expr, const = only_positive_bv_sub(flatexpr)
+    if isinstance(const, list):
```

Comment:
that is not possible per your doc of that function?

Reply:
or wait, it checks for const to be a list? what does that mean? there is some hidden (unmaintainable) logic here...